src/Log/LogService.php

```diff
@@ -0,0 +1,368 @@
+            $this->logFileName = $logFileName;
+        }
+
+        $logFilePath = $this->getPath().$this->logFileName;
+        $this->logFile = @fopen($logFilePath, 'a', false);
+        if ($this->logFile === false) {
+            $path = dirname($logFilePath);
+
+            if (! is_dir($path)) {
+                throw new Exception('Directory for logging does not exist');
+            } else {
+                throw new Exception('Failed to open logging file:' . $logFilePath);
+            }
+        }
+    }
+
+    /**
+     * @param null $logger \Zend_Log
+     * @return \Zend_Log
+     *
+     * To add log from external modules.
+     */
+    public function addLog($logger = null)		//should include addLog($name, $logger) -- REFACTOR
+    {
+        if ($logger == null) {
+            $logger = $this->createLogger($this->logFile);
+        	fclose($this->logFile);
+        }
+
+        $this->applyLogLevel($logger);
+
+        $this->loggers[$this->logFileName] = $logger;
+        return $logger;
+    }
+
+    /**
+     * @throws \Zend_Exception
+     */
+    public function getConfig()
+    {
+        $this->config = \Zend_Registry::get('Zend_Config');
+        return $this->config;
+    }
+
+    /**
+     * get path to the log folder.
+     *
+     * @return string path to the log folder.
+     */
+    public function getPath()
+    {
+        if ($this->logPath == null) {
+        	$config = $this->getConfig();
+        	$this->setpath($config->workspacePath . '/log/');
+        }	
+
+        return $this->logPath;	
+    }
+
+    /**
+     * set path to the log file.
+     *
+     * @param $logPath String optional to set the path.
+     *
+     * @return null path
+     */
+    public function setPath($logPath)
+    {
+        $this->logPath = $logPath;
+    }
+
+    /**
+     * Returns the default log
+     *
+     * @return mixed|Zend_Log
+     */
+    public function getDefaultLog()
+    {
+        if (array_key_exists($this->defaultLogName.'.log', $this->loggers)) {
+            return $logger = $this->loggers[$this->defaultLogName.'.log'];
+        } else {
+            return $this->createDefaultLog('default', 'opus.log');
+        }
+    }
+
+    /**
+     * @param $defaultLogName String
+     * @param $defaultLogFileName String
+     * @return \Zend_Log
+     *
+     *
+     * To set the default log
+     */
+    public function createDefaultLog($defaultLogName, $defaultLogFileName)
+    {
+        $this->defaultLogName = $defaultLogName;
+        $this->createLog($defaultLogName, $defaultLogFileName);
+        return $this->addLog();
+    }
+
+    /**
+     * @param $logName String
+     * @return null
+     *
+     * TODO
+     */
+    public function getLogConfig($logName)
+    {
+    	return null;		
+    }
+
+    /**
+     *
+     * @param $logger \Zend_Log object
+     * @throws \Zend_Exception
+     * @throws \Zend_Log_Exception
+     *
+     * Gets log level constants from Zend_Log and apply log levels.
```

Comment:
This function does multiple things. A description with an "and" is a warning. But besides that, the function determines the default log level, which by the way is the log level configured unless there is nothing configured or it was set for the LogService manually. So all this should go into getDefaultLogLevel(). The function also handles verifying the log level string. It then has a  filter object created and adds to a logger. 

Currently it also sets the LOG_LEVEL in the global registry and this will happen more than once, perhaps with different levels, which wasn't the original intent. So as it is, this will create a bug, because there is other code relying on LOG_LEVEL being set to the level of the default logger.